sklearn/utils/estimator_checks.py

```diff
@@ -1081,91 +1080,72 @@ def check_sample_weights_shape(name, estimator_orig):
 
 
 @ignore_warnings(category=FutureWarning)
-def check_sample_weights_invariance(name, estimator_orig, kind="ones"):
-    # For kind="ones" check that the estimators yield same results for
-    # unit weights and no weights
-    # For kind="zeros" check that setting sample_weight to 0 is equivalent
-    # to removing corresponding samples.
-    estimator1 = clone(estimator_orig)
-    estimator2 = clone(estimator_orig)
-    set_random_state(estimator1, random_state=0)
-    set_random_state(estimator2, random_state=0)
-
-    X1 = np.array(
-        [
-            [1, 3],
-            [1, 3],
-            [1, 3],
-            [1, 3],
-            [2, 1],
-            [2, 1],
-            [2, 1],
-            [2, 1],
-            [3, 3],
-            [3, 3],
-            [3, 3],
-            [3, 3],
-            [4, 1],
-            [4, 1],
-            [4, 1],
-            [4, 1],
-        ],
-        dtype=np.float64,
-    )
-    y1 = np.array([1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int)
+def check_sample_weights_invariance(name, estimator_orig):
+    # check that setting sample_weight to zero / integer is equivalent
+    # to removing / repeating corresponding samples.
+    estimator_weighted = clone(estimator_orig)
+    estimator_repeated = clone(estimator_orig)
+    set_random_state(estimator_weighted, random_state=0)
+    set_random_state(estimator_repeated, random_state=0)
 
-    if kind == "ones":
-        X2 = X1
-        y2 = y1
-        sw2 = np.ones(shape=len(y1))
-        err_msg = (
-            f"For {name} sample_weight=None is not equivalent to sample_weight=ones"
-        )
-    elif kind == "zeros":
-        # Construct a dataset that is very different to (X, y) if weights
-        # are disregarded, but identical to (X, y) given weights.
-        X2 = np.vstack([X1, X1 + 1])
-        y2 = np.hstack([y1, 3 - y1])
-        sw2 = np.ones(shape=len(y1) * 2)
-        sw2[len(y1) :] = 0
-        X2, y2, sw2 = shuffle(X2, y2, sw2, random_state=0)
-
-        err_msg = (
-            f"For {name}, a zero sample_weight is not equivalent to removing the sample"
-        )
-    else:  # pragma: no cover
-        raise ValueError
+    rng = np.random.RandomState(42)
+    n_samples = 15
+    X = rng.rand(n_samples, 2)
+    y = rng.randint(0, 2, size=n_samples)
+    # Use random integers (including zero) as weights.
+    sw = rng.randint(0, 5, size=n_samples)
+    # Check that at least one data point has zero weight
+    assert np.isin(0, sw)
```

Comment:
Something I've started to do, is to add a good explanation on all asserts in these checks to tell the user how to fix the issue. I think we should do the same here, and let them know what exactly is failing, and how they could fix it.

Reply:
do you mean something like:
```python
assert np.isin(0, sw), "make sure at least one data point has zero weight"
```

Reply:
@adrinjalali maybe I could simple remove these two assertions as they are not really part of the test API but an internal check when designing the test ?

Reply:
Alternatively we could have kept it and expanded the comment:

```python
    # Check that at least one data point has zero weight so has to test
    # that fitting zero weight data points is equivalent to dropping them.
    # Since the test data is random, this depends on the choice of the seed
    # and the value `n_samples`. Increasing `n_samples` makes this more likely
    # to have at least one zero-weight data point in the test data. However
    # increasing `n_samples` too much might make this test run slower and can
    # also cause the estimator to be less sensitive to changes in weights and
    # might as a result make this check more likely to fail to detect violations
    # of the weight/repetitions equivalence property we want to enforce in all
    # scikit-learn estimators.
    assert np.isin(0, sw)
```

Reply:
@adrinjalali does the comment above addresses what you were looking for in your original comment?

Reply:
My point here is that the common tests are basically developer API kinda thing. When people write a custom estimator and run estimator checks, they should get error messages that tell them how to fix their issues.

However, in this case, this assert should never be triggered since we're testing the data generated in the test. I like the comment you're suggesting. I'd add a message for the assert being like, "This shouldn't have happened, please open an issue on ...." kinda thing.

Reply:
@antoinebaker you can put this back with the comment suggested by Olivier.